security/pkg/nodeagent/cache/secretcache.go

```diff
@@ -161,9 +162,9 @@ func (sc *SecretCache) rotate(t time.Time) {
 	})
 }
 
-func (sc *SecretCache) generateSecret(token string, t time.Time) (*sds.SecretItem, error) {
+func (sc *SecretCache) generateSecret(proxy model.Proxy, token string, t time.Time) (*sds.SecretItem, error) {
 	options := util.CertOptions{
-		Host:       "", //TODO(quanlin): figure out what to use here.
+		Host:       proxy.IPAddress,
```

Comment:
The host should be the SPIFFE uri, definitely not the IP address. The SPIFFEE uri for GCP service account should be in form of the spiffe://accounts.google.com/<gcp service account>.
And we can configure the Envoy SDS's secret name to be this uri, so here you can just use the name from the Request.

+ @justinburke  to confirm.

Reply:
as mentioned in another comment, host here is used in csr template to generate cert in nodeagent locally; from [BuildSubjectAltNameExtension implementation](https://github.com/istio/istio/blob/97882cede142f5ac246a2e7cff4d6ff6bf631389/security/pkg/pki/util/san.go#L79), either service account or IP address should be fine. 

Since SDS is a general service, in theory, sds shouldn't assume envoy pass service account as secret name to server side, also t's not supported from client side now(@qiwzhang @JimmyCYJ )

Reply:
@incfly is correct that we don't want to be using IP addresses in the certificates. The SVIDs (ie SPIFFE certificates) will identify service accounts and not service locations (like IP addresses or host names). Note that this is a different authentication model that you're probably used to seeing: outside of Istio, we're used to seeing certificates for service locations (IP address, hostname), but within Istio, we'll be using service accounts as the basis for mutual authentication.